tasks/Task.cpp

```diff
@@ -156,35 +174,73 @@ gps_base::Solution Task::convertToBaseSolution(const PVT &data) const
             solution.positionType = gps_base::AUTONOMOUS_2D;
             break;
         case PVT::FIX_3D:
-        case PVT::GNSS_PLUS_DEAD_RECKONING:
-            solution.positionType = gps_base::AUTONOMOUS;
+        case PVT::GNSS_PLUS_DEAD_RECKONING: {
+            if (data.fix_flags & PVT::FIX_RTK_FIXED) {
+                solution.positionType = gps_base::RTK_FIXED;
+            }
+            else if (data.fix_flags & PVT::FIX_RTK_FLOAT) {
+                solution.positionType = gps_base::RTK_FLOAT;
+            }
+            else {
+                solution.positionType = gps_base::AUTONOMOUS;
+            }
             break;
+        }
         default:
             solution.positionType = gps_base::INVALID;
             break;
     }
     solution.time = data.time;
     return solution;
 }
-void Task::processIO()
-{
-    gps_ublox::UBX::Frame frame = mDriver->readFrame();
-    if (frame.msg_class == UBX::MSG_CLASS_NAV && frame.msg_id == UBX::MSG_ID_PVT) {
-        gps_ublox::PVT data = UBX::parsePVT(frame.payload);
 
-        _pose_samples.write(convertToRBS(data));
-        _gps_solution.write(convertToBaseSolution(data));
+struct gps_ublox::PollCallbacks : gps_ublox::Driver::PollCallbacks {
```

Comment:
I wasn't aware that we could use a qualified name on struct definition

Reply:
Neither was I ... just did it without thinking about it.